packages/medusa/src/api-v2/admin/campaigns/[id]/route.ts

```diff
@@ -17,3 +18,26 @@ export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
 
   res.status(200).json({ campaign })
 }
+
+export const POST = async (req: MedusaRequest, res: MedusaResponse) => {
+  const updateCampaigns = updateCampaignsWorkflow(req.scope)
+  const manager = req.scope.resolve("manager")
+  const campaignsData = [
+    {
+      id: req.params.id,
+      ...(req.validatedBody || {}),
+    },
+  ]
+
+  const { result, errors } = await updateCampaigns.run({
+    input: { campaignsData },
+    context: { manager },
+    throwOnError: false,
+  })
+
+  if (Array.isArray(errors) && errors[0]) {
+    throw errors[0].error
```

Comment:
question: Is this how we are supposed to handle possibly multiple errors?

cc @adrien @carlos-r-l-rodrigues

Reply:
From the last time we spoke, this was the discussed approach to keep the MedusaError context from the modules. I'm unsure if anything has changed since then. Let me know if I missed anything. 🤞🏻

Reply:
I would say that if there is multiple errors then we should return them all to the client such that the client doesn't first the first error and still receive another error for the next call. I suppose it would require some changes to the error handler for that to be supported. wdyt guys?

Reply:
The client of the workflow here is the API handler and the API handler needs to only throw the errors from the success paths. Any other error that pops up here is from the compensate flow, which shouldn't be returned to the client consuming the API.

Reply:
I would say from the workflow, I'd expect info about which errors are occurring where (compensate, success) and the consumer of the workflow can then decide what to do with it. All success errors, the API handler throws and all compensate errors will be logged to sentry or log. 

Currently the API can only handle single error, but with the workflows we could have instances where we're running 2 parallel steps that throw an error each. From the API handler perspective, we have to make this change and come up with a convention.

Reply:
@adrien2p do you have a suggestion for a different approach that would be easy to implement now? Otherwise, I think we should go with this and revisit later to not block this PR from being merged

Reply:
yes we can revisit later, but what about just not putting `throwOnError: false`?